src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateEntry.cs

```diff
@@ -27,5 +29,12 @@ public class ModelStateEntry
         /// Gets or sets the <see cref="ModelValidationState"/> for this entry.
         /// </summary>
         public ModelValidationState ValidationState { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="ModelStateEntry"/> for a sub-property with the specified <paramref name="propertyName"/>.
```

Comment:
We use in the ValidationHelper to locate subproperties. The alternative would be to construct the property name from `ModelMetadata` as we traverse the tree.